### ISSUE TITLE
incorrect experiment run URL generated from the mlflow library

### ISSUE THREAD

**Issue by happyroyce** [mlflow/mlflow#13204]
The URL of the experiment run printed in the logs is incorrect, pointing to the wrong host. This appears to originate from the MLFlow library. 

**Code reference:**

File: https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/_tracking_service/client.py#L903
line 903 and 904
`_logger.info(f"🏃 View run {run_name} at: {run_url}.")`
`_logger.info(f"🧪 View experiment at: {experment_url}.")`

File: https://github.com/mlflow/mlflow/blob/990b453e7632513adf60a73bd57715ba72407481/mlflow/utils/databricks_utils.py#L467
line 467
```
def _get_workspace_url():
    try:
        if spark_session := _get_active_spark_session():
            if workspace_url := spark_session.conf.get("spark.databricks.workspaceUrl", None):
                return workspace_url
    except Exception:
        return None
```

**Suggested Fix:**
The host URL logic in _client.py_ and _databricks_utils.py_ may need to be revisited to ensure it is generating the correct URLs based on the configuration of the MLflow tracking service.

**Comment by B-Step62** [comment]
@happyroyce Could you fill in the bug report issue template and share necessary information about your environment setup (e.g. where you run tracking server, host OS, proxy, etc) and exact issue in the printed URLs?

**Comment by happyroyce** [comment]
> @happyroyce Could you fill in the bug report issue template and share necessary information about your environment setup (e.g. where you run tracking server, host OS, proxy, etc) and exact issue in the printed URLs? We cannot diagnose the issue with the provided information.

Yep so we are running mlflow in databricks and we have two databricks workspaces and mlflow is hosted under a different databricks workspaces. We use the registry URI and some code samples about how we use those are as follows,
```
uri = f"databricks://{scope}:{prefix}"
mlflow.set_tracking_uri(uri)
mlflow.set_registry_uri(uri)
```
